### PR TITLE
Merge upstream tag v1.2.0 

### PR DESCRIPTION
**Pull request by jmrodri**
Merge upstream tag v1.2.0 

    Operator SDK v1.2.0

    Merge executed via ./UPSTREAM-MERGE.sh v1.2.0 upstream master

    Overwritten conflicts:
    .gitignore
    Makefile

### CODE REVIEW

**Comment by jmrodri** [comment]
you can extract the operator-sdk binary from the sdk image built by the PR, for example, 
```
docker create registry.build01.ci.openshift.org/ci-op-nkc3fx06/stable:operator-sdk # returns a SHA

# use the SHA from above to reference the container
docker cp 2270a139c02a5e3d43cc0be735622a5466267ad01d76fd0d0c938ce3c0fd3ff8:/user/local/bin/operator-sdk /tmp/operator-sdk
```
you will now have the `operator-sdk` binaries

File: images/ansible-operator/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+# yum clean all && rm -rf /var/yum/cache/* first
+RUN yum clean all && rm -rf /var/cache/yum/* \
+  && yum -y update \
+  && yum install -y libffi-devel openssl-devel python36-devel gcc python3-pip python3-setuptools \
+  && pip3 install --no-cache-dir \
+    ipaddress \
+    ansible-runner==1.3.4 \
+    ansible-runner-http==1.0.0 \
+    openshift~=0.10.0 \
+    ansible~=2.9 \
```
Comment (estroz) [inline images/ansible-operator/Dockerfile:24]: I know we're just merging commit wholesale but this needs to be pinned to 2.9.15 (see https://github.com/operator-framework/operator-sdk/pull/4321). We can either wait for v1.2.1 or v1.3. Thoughts @jmrodri?

/ping @asmacdo

Comment (jmrodri) [reply images/ansible-operator/Dockerfile:24]: Those aren't being used downstream. I probably need to compare the files on release/ansible

**Comment by emmajiafan** [comment]
@jmrodri  I check the operatorsdk version is :./operator-sdk version
operator-sdk version: "9762a774", commit: "9762a774472487cd72784cf0e591e6cf44fc09d4", kubernetes version: "v1.18.8", go version: "go1.15.5", GOOS: "linux", GOARCH: "amd64". It is not "ocp4.7" or "1.2.0". So is it the expected result?

File: images/ansible-operator/Dockerfile (hunk 1 above)
Comment (jmrodri) [reply images/ansible-operator/Dockerfile:24]: @estroz I've compared the 2. The downstream has its own dockerfiles https://github.com/openshift/ocp-release-operator-sdk/tree/master/release I usually just compare them to see if there's anything significant that needs to be brought over. The versions of ansible etc, are all controlled by the packages in the downstream repo which are up to date.

Comment (estroz) [reply images/ansible-operator/Dockerfile:24]: Cool, resolved.

**Review by estroz — APPROVED** [review]
/lgtm

**Comment by jmrodri** [comment]
@emmajiafan not sure why the version isn't getting set. I updated the dockerfiles in this PR to set it directly. We could write a bug against it and address it later. I feel like this PR might be using cached images. Please try to test what you can

**Comment by emmajiafan** [comment]
> @emmajiafan not sure why the version isn't getting set. I updated the dockerfiles in this PR to set it directly. We could write a bug against it and address it later. I feel like this PR might be using cached images. Please try to test what you can
Ok, will open a bug to trace it. And this pr merge a lot of codes ,so I will do a regression test about it and will cost about 2 days.

**Comment by emmajiafan** [comment]
@jmrodri  Ok, will open a bug to trace it. And this pr merge a lot of codes ,so I will do a regression test about it and will cost about 2 days.

**Comment by jmrodri** [comment]
@emmajiafan that is expected. It's a full 1.2.0 release from upstream.

**Pull request merged**